src/sentry/integrations/opsgenie/utils.py

```diff
@@ -1,13 +1,87 @@
+import logging
 from typing import Optional
 
+from sentry.constants import ObjectStatus
+from sentry.incidents.models import AlertRuleTriggerAction, Incident, IncidentStatus
+from sentry.integrations.metric_alerts import incident_attachment_info
+from sentry.services.hybrid_cloud.integration import integration_service
 from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+from sentry.shared_integrations.exceptions import ApiError
+
+logger = logging.getLogger("sentry.integrations.opsgenie")
+from .client import OpsgenieClient
+
+
+def build_incident_attachment(incident, new_status: IncidentStatus, metric_value=None):
+    data = incident_attachment_info(incident, new_status, metric_value)
+    alert_key = f"incident_{incident.organization_id}_{incident.identifier}"
+    if new_status == IncidentStatus.CLOSED:
+        payload = {"identifier": alert_key}
+    priority = "P1"
+    if new_status == IncidentStatus.WARNING:
+        priority = "P2"
+    else:
+        payload = {
+            "message": incident.alert_rule.name,
+            "alias": alert_key,
+            "description": data["text"],
+            "source": "Sentry",
+            "priority": priority,
+            "details": {
+                "URL": data["title_link"],  # type: ignore
+            },
+        }
+    return payload
```

Comment:
Is it possible to add types to this function and the call to `incident_attachment_info` so we don't have to type ignore? If not, this seems reasonable but I want to check.

Reply:
It seems like it's not possible due to alert_rule.organization being a FlexibleForeignKey. I've added typing to the provided functions, though.